modules/oauth/src/main/java/org/mule/module/oauth2/internal/authorizationcode/state/ConfigOAuthContext.java

```diff
@@ -41,7 +41,9 @@ public ResourceOwnerOAuthContext getContextForResourceOwner(final String resourc
         if (!oauthContextStore.containsKey(resourceOwnerId))
         {
             final Lock lock = lockFactory.createLock(configName + "-config-oauth-context");
+            final Lock resourceLock = createLockForResourceOwner(resourceOwnerId);
             lock.lock();
```

Comment:
Shouldn't it be better to use 2 try-finally blocks? 
If the first lock was acquired and the second fails for some reason, the finally block will never be executed and the first lock will remain acquired.